src/app/components/layout/app-layout/app-layout.component.ts

```diff
@@ -140,15 +141,14 @@ export class AppLayoutComponent implements OnInit, OnDestroy {
     this.router.events
       .pipe(
         filter(event => {
-          return (
-            event instanceof NavigationStart &&
-            // search employs reuse route strategy
-            !event.url.startsWith('/search;')
-          );
+          return event instanceof NavigationStart;
         }),
         takeUntil(this.onDestroy$)
       )
-      .subscribe(() => this.store.dispatch(new SetSelectedNodesAction([])));
+      .subscribe(() => {
+        this.store.dispatch(new SetSelectedNodesAction());
+        this.store.dispatch(new ReloadDocumentListAction());
```

Comment:
I have realized that this reload should not be done on each route change. Maybe we could trigger it for '/search' only on toggle-favorite in a similar way as we do on '/favorites' and '/favorite/libraries'

Reply:
yes, I know

Reply:
as agreed, we will raise a separate ticket